packages/skeleton-react/src/lib/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,196 @@
+import { IconProps, RatingContextState, RatingProps } from "./types";
+
+// Context
+
+export const RatingContext = createContext<RatingContextState>({
+    interactive: false,
+    value: 0,
+    order: 0,
+    figureRef: undefined
+});
+
+// Components ---
+const RatingRoot: React.FC<RatingProps> = ({
+    value = 0,
+    max = 5,
+    interactive = false,
+    fraction = 1,
+    // Root
+    base = 'flex',
+    width = 'w-full',
+    text = 'text-token',
+    fill = 'fill-token',
+    justify = 'justify-center',
+    spaceX = 'space-x-2',
+    classes = '',
+    // Item ---
+    buttonBase = 'w-full h-full',
+    buttonPosition = 'relative',
+    buttonAspect = 'aspect-square',
+    buttonClasses = '',
+    // Events
+    onMouseDown = () => {},
+    onKeyDown = () => {},
+    onValueChange = () => {},
+	// Children
+	children,
+}) => {
+    const figureRef = useRef<HTMLElement>(null);
+
+    const [focusedButtonIndex, setFocusedButtonIndex] = useState(0);
+    useEffect(() => {
+        const index = Math.max(0, Math.ceil(value - 1));
+        setFocusedButtonIndex(index);
+    }, [value]);
+
+    const onRatingMouseDown = useCallback((event: React.MouseEvent<HTMLButtonElement>, order: number) => {
+        if(!figureRef.current) return;
+
+        const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+        const fractionWidth = ratingRect.width / fraction;
+        const left = event.clientX - ratingRect.left;
+        let selectedFraction = Math.floor(left / fractionWidth) + 1;
+
+        if(getComputedStyle(figureRef.current).direction === 'rtl') {
+            selectedFraction = fraction - selectedFraction + 1;
+        }
+
+        value = order + selectedFraction / fraction;
+        onValueChange(value);
+        onMouseDown(event, value);
+    }, [fraction]);
+
+    const onRatingKeyDown = useCallback((event: React.KeyboardEvent<HTMLButtonElement>) => {
+        if(!figureRef.current) return;
+
+        const rtl = getComputedStyle(figureRef.current).direction === 'rtl';
+        switch(event.key) {
+            case 'ArrowLeft':
+                rtl ? increaseValue() : decreaseValue();
+                break;
+            case 'ArrowRight':
+                rtl ? decreaseValue() : increaseValue();
+                break;
+        }
+        onKeyDown(event);
+    }, []);
+
+    function refreshFocus() {
+        if(!figureRef.current) return;
+
+        const buttons = figureRef.current.querySelectorAll('button');
+        buttons[Math.max(0, Math.ceil(value - 1))].focus();
+    }
+
+    function increaseValue() {
+        value = Math.min(max, value + 1 / fraction);
+        onValueChange(value);
+        refreshFocus();
+    }
+
+    function decreaseValue() {
+        value = Math.max(0, value - 1 / fraction);
+        onValueChange(value);
+        refreshFocus();
+    }
+
+    return(
+        <>
+            <style>{`
```

Comment:
Unless React has a specific implementation, I'd recommend keeping `<style>` blocks at the bottom of the template to match Svelte, Vue, etc

Reply:
Per the solution suggested above, there is no `style` block anymore :partying_face: